File: src/main/java/sprykerFileCreator/model/renderer/dto/PhpClassItem.java
package sprykerFileCreator.model.renderer.dto;

import com.intellij.openapi.util.text.StringUtil;
import com.sun.deploy.util.StringUtils;

import java.util.ArrayList;
import java.util.List;

public class PhpClassItem implements PhpClassInterface {

    private String namespace;
    private String name;
    private Boolean isAbstract;
    private PhpClassInterface parentClass;
    private String parentAlias = "";
    private List<UseBlockItem> useBlockItems;
    private List<DocBlockItem> docBlockItems;

    public PhpClassItem(String namespace, String className) {
        this.namespace = namespace;
        this.name = className;
        this.useBlockItems = new ArrayList<UseBlockItem>();
        this.docBlockItems = new ArrayList<DocBlockItem>();
    }

    @Override
    public String getNamespace() {
        return this.namespace;
    }

    @Override
    public String getName() {
        return this.name;
    }

    @Override
    public String getFullQualifiedName() {

        return StringUtil.trimEnd(this.namespace, "\\") + "\\" + this.name;
    }

    @Override
    public Boolean isAbstract() {
        return this.isAbstract;
    }

    @Override
    public void setAbstract() {
        this.isAbstract = true;
    }

    @Override
    public PhpClassInterface getParentClass() {
        return this.parentClass;
    }

    @Override
    public void setParentClass(PhpClassInterface parentClass) {
        this.setParentClass(parentClass, "Core");
    }

    @Override
    public void setParentClass(PhpClassInterface parentClass, String aliasPrefix) {
        String parentClassName = parentClass.getName();
        String parentNamespace = parentClass.getNamespace();
        parentNamespace = StringUtil.trimStart(parentNamespace, "\\");
        UseBlockItem parentUseItem = new UseBlockItem(parentClassName, parentNamespace);

        if (parentClassName.equals(this.name)) {
//            if (parentNamespace.equals(this.namespace)) {
//                throw new Exception();
//            }
            this.parentAlias = aliasPrefix + parentClassName;
            parentUseItem.setAlias(this.parentAlias);
        }

        this.addUseBlockItem(parentUseItem);
        this.parentClass = parentClass;
    }

    @Override
    public List<UseBlockItem> getUseBlockItems() {
        return this.useBlockItems;
    }

    @Override
    public void addUseBlockItem(UseBlockItem useBlockItem) {
        this.useBlockItems.add(useBlockItem);
    }

    @Override
    public List<DocBlockItem> getDocBlockItems() {
        return this.docBlockItems;
    }

    @Override
    public void addDocBlockItem(DocBlockItem docBlockItem) {
        this.docBlockItems.add(docBlockItem);
    }

    @Override
    public String getParentAlias() {
        return this.parentAlias;
    }

}

File: src/main/kotlin/com/github/tobi812/sprykerplugin/models/renderer/dto/PhpClassItem.kt
package com.github.tobi812.sprykerplugin.models.renderer.dto

import com.intellij.openapi.util.text.StringUtil
import java.util.ArrayList

class PhpClassItem(override val namespace: String, override val name: String) : PhpClassInterface {
    override var isAbstract: Boolean = false
    override var parentClass: PhpClassInterface? = null
    override var parentAlias = ""
    override val useBlockItems: MutableList<UseBlockItem>
    override val docBlockItems: MutableList<DocBlockItem>

    init {
        this.useBlockItems = ArrayList()
        this.docBlockItems = ArrayList()
    }

    override fun getFullQualifiedName(): String {
        return StringUtil.trimEnd(this.namespace, "\\") + "\\" + this.name
    }

    override fun setAbstract(): PhpClassInterface {
        this.isAbstract = true

        return this
    }

    override fun setParentClass(parentClass: PhpClassInterface): PhpClassInterface {
        return this.setParentClass(parentClass, "Spryker")
    }

    override fun setParentClass(parentClass: PhpClassInterface, aliasPrefix: String): PhpClassInterface {
        val parentClassName = parentClass.name
        var parentNamespace = parentClass.namespace
        parentNamespace = StringUtil.trimStart(parentNamespace, "\\")
        val parentUseItem = UseBlockItem(parentClassName, parentNamespace)

        if (parentClassName == name) {
//            if (parentNamespace.equals(this.namespace)) {
//                throw new Exception();
//            }
            parentAlias = aliasPrefix + parentClassName
            parentUseItem.alias = parentAlias
        }

        this.addUseBlockItem(parentUseItem)
        this.parentClass = parentClass

        return this
    }

    override fun addUseBlockItem(useBlockItem: UseBlockItem): PhpClassInterface {
        this.useBlockItems.add(useBlockItem)

        return this
    }

    override fun addDocBlockItem(docBlockItem: DocBlockItem): PhpClassInterface {
        this.docBlockItems.add(docBlockItem)

        return this
    }
}